### PR TITLE
Added commands reset arenaworld and honorworld

### PR DESCRIPTION
**Pull request by Helias**
Added "reset arenaseason" and "reset honorworld" commands.

I took this commands from here: https://github.com/WarHead/uwom-server/blob/691b62a0054324e1e57405ff827c4dc6393d3bec/src/server/scripts/Commands/cs_reset.cpp
and I modified and translated it.

Now the commands do the following things:

**.reset arenaseason**
reset all the arena team rating and the seasonGames/Wins and weekGames/Wins at the default value
and with the [delete] parameter delete also all the arena teams.

**.reset honorpoints**
reset at the default values the honorpoints and with the parameter [kills] also the kills

They works but:
.reset honorworld works only for non online players
To see the results of both commands the server needs a restart because the commands modify some tables that we can't reload just with the command .reload name_table so we need to restart the server
(unfortunately the command .reload can't reload all the tables and unfortunately the tables that are handled with these commands can't be reloaded).

### CODE REVIEW

File: sql/updates/world/world_command.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+DELETE FROM `command` WHERE `name`='reset arenaworld';
+DELETE FROM `command` WHERE `name`='reset honorworld';
+
+INSERT INTO `command` (`name`, `permission`, `help`) VALUES
+('reset honorworld', 900, 'Syntax: .reset honorworld [kills]\nReset all honor points at the default values. With [kills] all the kills will be resetted.'),
+('reset arenaworld', 901, 'Syntax: .reset arenaworld [delete]\nReset all arena points/rating at the default values. With [delete] all teams will be deleted.');
```
Comment (MitchesD) [inline sql/updates/world/world_command.sql]: no new line

**Comment by Helias** [comment]
New line added to world_command.sql

**Comment by Helias** [comment]
Changed the names of the commands and improved the codestyle under suggestions of @MitchesD ;)

File: src/server/game/Accounts/RBAC.h [hunk 2]
```diff
@@ -697,6 +697,10 @@ enum RBACPermissions
     // 799 - 834 6.x only
     RBAC_PERM_COMMAND_DEBUG_LOADCELLS                        = 835,
 
+
+    RBAC_PERM_COMMAND_RESET_HONORWORLD                       = 900,
+    RBAC_PERM_COMMAND_RESET_ARENASEASON                       = 901,
```
Comment (MitchesD) [inline src/server/game/Accounts/RBAC.h]: fix codestyle :D

File: src/server/scripts/Commands/cs_reset.cpp [hunk 3]
```diff
@@ -39,6 +39,8 @@ class reset_commandscript : public CommandScript
     {
         static ChatCommand resetCommandTable[] =
         {
+            { "honorworld",   rbac::RBAC_PERM_COMMAND_RESET_HONORWORLD,  true, &HandleResetHonorWorldCommand,   "", NULL },
+            { "arenaseason",   rbac::RBAC_PERM_COMMAND_RESET_ARENASEASON,  true, &HandleResetArenaSeasonCommand,   "", NULL },
```
Comment (MitchesD) [inline src/server/scripts/Commands/cs_reset.cpp]: wrong codestyle too

**Comment by MitchesD** [comment]
also you should update auth_base to fix buid errors

**Comment by Nyeriah** [comment]
What happens if someone runs this while an arena match is in progress?

**Comment by Helias** [comment]
I didn't test this

File: src/server/scripts/Commands/cs_reset.cpp [hunk 4]
```diff
@@ -39,6 +39,8 @@ class reset_commandscript : public CommandScript
     {
         static ChatCommand resetCommandTable[] =
         {
+            { "honorworld",   rbac::RBAC_PERM_COMMAND_RESET_HONORWORLD,  true, &HandleResetHonorWorldCommand,   "", NULL },
```
Comment (MitchesD) [inline src/server/scripts/Commands/cs_reset.cpp]: still wrong codestyle, check other lines

**Comment by Demonid** [comment]
@Nyeriah I believe these commands will be used after ending a arena season (Disable Arenas via config)

**Comment by Helias** [comment]
I reopened the pull to fix a merge conflict https://github.com/TrinityCore/TrinityCore/pull/15679

**Pull request closed without merging**